sql/tablespace.sql

```diff
@@ -38,23 +38,16 @@ BEGIN
     INTO dimension_id, partitioning_func;
 
     -- Find all dimension slices for the chosen dimension
-    dimension_slices := ARRAY(
+    hypertable_dimension_slices := ARRAY(
         SELECT s.id FROM _timescaledb_catalog.dimension_slice s
         WHERE (s.dimension_id = main_block.dimension_id)
+        ORDER BY s.id
     );
 
-    -- Find the chunk's dimension slice for the chosen dimension
-    SELECT s.id FROM _timescaledb_catalog.dimension_slice s
-    INNER JOIN _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = s.id)
-    INNER JOIN _timescaledb_catalog.chunk c ON (cc.chunk_id = c.id)
-    WHERE (s.dimension_id = main_block.dimension_id)
-    AND (c.id = select_tablespace.chunk_id)
-    INTO STRICT chunk_slice_id;
-
     -- Find the array index of the chunk's dimension slice
     SELECT i
-    FROM generate_subscripts(dimension_slices, 1) AS i
-    WHERE dimension_slices[i] = chunk_slice_id
+    FROM generate_subscripts(hypertable_dimension_slices, 1) AS i
+    WHERE hypertable_dimension_slices[i] = ANY(chunk_dimension_slice_ids)
```

Comment:
Can you clarify in a comment that `hypertable_dimension_slices[]` contains only slices from the same dimension while `chunk_dimension_slice_ids[]` contains exactly one slice from each dimension (this is what is called a hypercube in the C code for clarity. Also note the inconsistent naming of the arrays where one indicates slice objects (`_slices[]`) and one indicates IDs (`_slice_ids[]`). Both are really IDs.

Reply:
fixed